### PR TITLE
builder: Docker installation

### PR DESCRIPTION
**Pull request by sodar**
- Removed PSCX installation since Ansible _promises_ that `.zip` archives are supported natively:
    - https://docs.ansible.com/ansible/2.4/win_unzip_module.html#synopsis
- Add Docker installation to builder role. Used in nightly builds.
- Add playbook for updating builders.

TO DO:

- [x] Create a builder template to test PSCX installation

### CODE REVIEW

**Comment by sodar** [comment]
Started `create-builder-template` job: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-builder-template/121/

**Comment by sodar** [comment]
Started new `create-builder-template` job: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-builder-template/122/

File: ansible/roles/builder/tasks/main.yml [hunk 1]
```diff
@@ -12,6 +12,63 @@
...
+  register: hyperv_installation
+  tags:
+    - builder_docker
+
+- name: Reboot the system
+  win_reboot:
+    test_command: powershell.exe 'Get-WindowsFeature'
+    pre_reboot_delay: 15
+    post_reboot_delay: 15
+  when: hyperv_installation is changed
```
Comment (michal-clapinski) [inline ansible/roles/builder/tasks/main.yml]: please use `reboot_required` instead

File: ansible/roles/builder/tasks/main.yml [hunk 2]
```diff
@@ -12,6 +12,63 @@
...
+  tags:
+    - builder_docker
+
+- name: Make sure Docker has started
+  win_service:
+    name: Docker
+    state: started
+    start_mode: auto
+  tags:
+    - builder_docker
```
Comment (michal-clapinski) [inline ansible/roles/builder/tasks/main.yml]: I think this is a leftover tag and should be removed

File: ansible/roles/builder/tasks/main.yml [hunk 3]
```diff
@@ -12,6 +12,61 @@
...
+- name: Install DockerMsftProvider
+  win_psmodule:
+    name: DockerMsftProvider
+  tags:
+    - builder_docker
+
+- name: Install Docker-EE
+  win_shell: "Install-Package -Name Docker -ProviderName DockerMsftProvider -Force"
+  tags:
+    - builder_docker
```
Comment (michal-clapinski) [inline ansible/roles/builder/tasks/main.yml]: I meant removing this tag from everywhere

Comment (sodar) [reply ansible/roles/builder/tasks/main.yml]: I think we should leave the tags, so this role can be used to install Docker on other builders without modifying it.

Comment (sodar) [reply ansible/roles/builder/tasks/main.yml]: In that case, tag should also be on `Make sure Docker has started` task.

Comment (michal-clapinski) [reply ansible/roles/builder/tasks/main.yml]: But will we ever again use this tag? If it's for one time usage only then I don't think it should be commited to repo

Comment (sodar) [reply ansible/roles/builder/tasks/main.yml]: Probably not. Ok, I'll remove the tags.

File: ansible/roles/builder/tasks/main.yml [hunk 4]
```diff
@@ -12,6 +12,48 @@
...
+    include_management_tools: true
+    state: present
+  register: hyperv_installation
+
+- name: Reboot the system
+  win_reboot:
+    test_command: powershell.exe 'Get-WindowsFeature'
+    pre_reboot_delay: 15
+    post_reboot_delay: 15
+  when: hyperv_installation.reboot_required
```
Comment (michal-clapinski) [inline ansible/roles/builder/tasks/main.yml]: I think it should be `hyperv_installation.feature_result.reboot_required`. What do you think?

File: ansible/roles/builder/tasks/main.yml [hunk 5]
```diff
@@ -12,6 +12,48 @@
...
+- name: Install Hyper-V feature
+  win_feature:
+    name: Hyper-V
+    include_management_tools: true
+    state: present
+  register: hyperv_installation
+
+- name: Reboot the system
+  win_reboot:
+    test_command: powershell.exe 'Get-WindowsFeature'
```
Comment (michal-clapinski) [inline ansible/roles/builder/tasks/main.yml]: why?

**Comment by sodar** [comment]
recheck no bug

File: ansible/roles/builder/tasks/main.yml [hunk 6]
```diff
@@ -12,6 +12,47 @@
...
+    state: present
+  register: hyperv_installation
+
+- name: Reboot the system
+  win_reboot:
+    pre_reboot_delay: 15
+    post_reboot_delay: 15
+  when: hyperv_installation.reboot_required or hyperv_installation.feature_result.reboot_required
+  tags:
+    - skip_ansible_lint
```
Comment (michal-clapinski) [inline ansible/roles/builder/tasks/main.yml]: why?

Comment (sodar) [reply ansible/roles/builder/tasks/main.yml]: Mindless copy paste from `testbed` role. Sorry. Removed the tags from `builder` and `testbed` roles and linter has passed.

**Comment by sodar** [comment]
`create-builder-template` from the latest commit in this PR: http://10.84.12.26:8080/job/WinContrail-infra/job/templates/job/create-builder-template/126/console

Builder template from the latest commit in this PR: `Template-builder-201809120909`

**Pull request merged**